tests/ses/nodes_preparation.pm

```diff
@@ -21,49 +21,60 @@ use utils 'systemctl';
 sub run {
     select_console 'root-console';
     # configure and start ntp, ntp server for nodes is master
-    if (check_var('NODE_HOSTNAME', 'master')) {
+    if (check_var('HOSTNAME', 'master')) {
         my $num_nodes = get_var('NODE_COUNT');
         barrier_create('network_configured', $num_nodes + 1);
+        barrier_create('deployment_done',    $num_nodes + 1);
         barrier_create('all_tests_done',     $num_nodes + 1);
         assert_script_run 'echo \'server ntp1.suse.de burst iburst\' >> /etc/ntp.conf';
     }
     else {
-        assert_script_run 'echo \'server master.openqa.de burst iburst\' >> /etc/ntp.conf';
+        assert_script_run 'echo \'server master.openqa.test burst iburst\' >> /etc/ntp.conf';
     }
+    # print the ntp config
+    assert_script_run 'grep -v ^# /etc/ntp.conf';
     systemctl 'restart ntpd';
     # disable ipv6
     assert_script_run 'echo \'net.ipv6.conf.all.disable_ipv6 = 1\' >> /etc/sysctl.conf';
+    systemctl 'restart network' unless get_var('DEEPSEA_TESTSUITE');
     # firewall and apparmor should not run
     systemctl 'stop SuSEfirewall2';
     systemctl 'disable SuSEfirewall2';
     systemctl 'stop apparmor';
     systemctl 'disable apparmor';
-    # set node hostname
-    my $node_hostname = get_var('NODE_HOSTNAME');
-    assert_script_run "hostnamectl set-hostname $node_hostname";
-    # configure network
-    configure_default_gateway;
-    my $node_ip = get_var('NODE_IP');
-    configure_static_ip("$node_ip/24");
-    configure_static_dns(get_host_resolv_conf());
-    # add node entries to /etc/hosts
-    my $hosts = <<'EOF';
-echo -e '10.0.2.100\tmaster.openqa.de master' >> /etc/hosts
-echo -e '10.0.2.101\tnode1.openqa.de node1' >> /etc/hosts
-echo -e '10.0.2.102\tnode2.openqa.de node2' >> /etc/hosts
-echo -e '10.0.2.103\tnode3.openqa.de node3' >> /etc/hosts
-echo -e '10.0.2.104\tnode4.openqa.de node4' >> /etc/hosts
+    if (get_var('DEEPSEA_TESTSUITE')) {
+        # set node hostname
+        my $node_hostname = get_var('HOSTNAME');
+        assert_script_run "hostnamectl set-hostname $node_hostname";
+        # configure network
+        configure_default_gateway;
+        my $node_ip = get_var('NODE_IP');
+        configure_static_ip("$node_ip/24");
+        configure_static_dns(get_host_resolv_conf());
+        # add node entries to /etc/hosts
+        my $hosts = <<'EOF';
```

Comment:
You use supportserver(dhcp,dns) - you can get hostname / IP from there. We use this in CaaSP to workaround hostnames that are set later: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/caasp/first_boot.pm#L49

Reply:
DEEPSEA_TESTSUITE does not use support_server
I don't see any benefit of adding another worker only because of network setup and we don't have thousands of workers that are idling